### ISSUE TITLE
Cant connect to voice chat if not on port 24454 (Ports open at UDP)

### ISSUE THREAD

**Issue by MrR341-GER** [henkelmax/simple-voice-chat#125]
**Please read the [FAQ](https://modrepo.de/minecraft/voicechat/faq) before submitting a bug report!**

**Bug description**

Cant start it on more server than just one.
I opened ports 24456 - 24460 UDP.
I set every Server on another Port within this range.
Server just starting fine, but the client can't connect to the voice server.

IP is on each properties file:
0.0.0.0

BUT it's working, if I set the port on 24454. But I can't run every server with this port in the voice chat properties

**Steps to reproduce the issue**

1. just multiple minecraft servers running

**Expected behavior**

Working voice-Chat

**Log files**
Please provide log files of the game session in which the problem occurred.
Don't paste the complete logs into the issue. 
You can use [https://gist.github.com/](https://gist.github.com/).

no logs

**Versions**
- Minecraft version
- Forge version
- Mod version

MC 1.16.5
Fabric 11.3
VC 1.1.1

**Other mods**
A list of the other mods that were installed when this problem occurred.
It would be ideal if you could test if this problem persists when no other mod is installed.
If not, please narrow down the mod(s) that are causing the problem.
Bungee,
architectury,
carpet,
Fabric-Proxy
lithium,
textile-backup,
worldedit

(none of them should be responsible for this issue (mostly becouse it works with port 24454 only))

**Screenshots (Optional)**
Screenshots of the issue.

**Comment by henkelmax** [comment]
Have you changed the port in the config?
If yes, olease send me the logs of the server and client.

**Comment by MrR341-GER** [comment]
Server (Each one on a different port):

[20:17:57] [VoiceChatServerThread/INFO]: Server started at port 24459

client in a minute

**Comment by MrR341-GER** [comment]
NOT WORKING Server:

[21:06:47] [main/INFO]: Connecting to server: 'mc.minology.de:24459'
[21:06:47] [main/INFO]: Setting sample rate to 48000 Hz, codec to VOIP and frame size to 1920 bytes
[21:06:47] [main/WARN]: Unknown custom packed identifier: carpet:hello
[21:06:47] [VoiceChatAuthenticationThread/INFO]: Trying to authenticate voice connection
[21:06:47] [main/INFO]: Loaded 15 advancements
[21:06:48] [VoiceChatAuthenticationThread/INFO]: Trying to authenticate voice connection

WOKING Server:
[21:08:27] [main/INFO]: Connecting to server: 'mc.minology.de:24454'
[21:08:27] [main/INFO]: Setting sample rate to 48000 Hz, codec to VOIP and frame size to 1920 bytes
[21:08:27] [main/WARN]: Unknown custom packed identifier: carpet:hello
[21:08:27] [VoiceChatAuthenticationThread/INFO]: Trying to authenticate voice connection
[21:08:27] [main/INFO]: Loaded 315 advancements
[21:08:27] [VoiceChatClientThread/INFO]: Server acknowledged authentication

**Comment by henkelmax** [comment]
Please do it like the template states and put the  FULL logs into a github gist link.

**Comment by MrR341-GER** [comment]
I don't know how to make a new gist... I already have one for VS-Code-Sync and can't create another gist
Obviously I'm too stupid

**Comment by MrR341-GER** [comment]
Oh yeah.. I am... okay gimme some sec

**Comment by MrR341-GER** [comment]
But I'm just posting the client, the working server and just one other not working server

**Comment by MrR341-GER** [comment]
Not working Server:

https://gist.github.com/MrR341-GER/efbc9fc419521603a24aa7ff3f454034

**Comment by MrR341-GER** [comment]
Working Server:

https://gist.github.com/MrR341-GER/7e1650eadb1a7a5d6ce3062abec5dbdc

**Comment by MrR341-GER** [comment]
Client joining on working Server:
https://gist.github.com/MrR341-GER/26c72b4e83b6c8dce129be4237117f96

**Comment by henkelmax** [comment]
Please also make sure to join with a client in the logs

**Comment by MrR341-GER** [comment]
Client joining on non working server:
https://gist.github.com/MrR341-GER/1a70c2a83b711cdb26bc1a3db13e471e

**Comment by MrR341-GER** [comment]
> Please also make sure to join with a client in the logs

jup... its in the comment of the gists

**Comment by henkelmax** [comment]
That's not the purpose of comments.
I need to be able to view it in raw, which I can't do with comments.

**Comment by MrR341-GER** [comment]
let me edit it
one sec

**Comment by MrR341-GER** [comment]
done

**Comment by henkelmax** [comment]
Does this happen without any other mods installed?

**Comment by MrR341-GER** [comment]
jup

**Comment by henkelmax** [comment]
Oh I just saw, you are using BungeeCord, right?

**Comment by MrR341-GER** [comment]
Yes

**Comment by henkelmax** [comment]
Yaeh that's not gonna work. How should the voice chat figure out if it changed the server.

**Comment by MrR341-GER** [comment]
Noo... It isn't installed on Bungee... it is installed on every Server on it's own

**Comment by MrR341-GER** [comment]
every server has it's own Voice Chat

**Comment by henkelmax** [comment]
I know. That will not work since bingee acts as a proxy.

**Comment by MrR341-GER** [comment]
But it is working wit bungee... but only at port 24454
and  bungee doesn't know anything about that... ant still.. it works

**Comment by henkelmax** [comment]
Yeah it only works with the first server, the client connects to.
Besides that, the mod uses cistom login packets, that most likely will not work wenn switching a server.

**Comment by MrR341-GER** [comment]
Cant confirm that... If I switch the port from the first server the client connects from 24454 to another one it isnt working...

**Comment by henkelmax** [comment]
I don't know what exactly doesn't work, bit I can assure you that BungeeCord is not working with this mod.

**Comment by MrR341-GER** [comment]
Well... testet without bungee.. still not working

**Comment by MrR341-GER** [comment]
completely without bungee:

**Comment by MrR341-GER** [comment]
completely without bungee on port 24459:

Server:
https://gist.github.com/MrR341-GER/9c74a5228c52ddf1ce5cf78b33f0925a

Client:
https://gist.github.com/MrR341-GER/3b3487a3b0f4d36628a2f5074ac34452

**Comment by henkelmax** [comment]
Seems that your port forwarding setup is wrong. Read the wiki for more information.
But even if it would work, it won't work with bungee.

**Comment by MrR341-GER** [comment]
Well...
I dont know what I did...
I just deleted the forewarding rule in my router and set it up again...
It's working now..
But:
EVEN WITH Bungee

**Comment by MrR341-GER** [comment]
Just to clarify why it doesnt matter if Bungee is running:

There isn't just one voice chat server running.

There are 4 different voice chat server running each on a different port.

**Comment by henkelmax** [comment]
Yes, but once you switch to another server with bungee, the voice chat will get confused and won't work like it should.

**Comment by MrR341-GER** [comment]
Just tested it... everything works perfect... I made a pic anyways :D

![image](https://user-images.githubusercontent.com/55445114/112618097-16f52c80-8e26-11eb-9c15-be16e6af29b9.png)

And if it is the client which you mean with "confused Voice Chat" then you could just add something to the code like:

```
If ($newConnection) {
    disable voice Chat
    flush Cache
    enable Voice Chat
}
```

**Comment by henkelmax** [comment]
I can't detect that in the mod, since the minecraft client itself doesn't even know that it moved to another server.
If you run a minecraft server on another machine than your bungee server, it will break too.

So as I already said. I can't guarantee you that it will work all the time.

**Comment by MrR341-GER** [comment]
Dont get me wrong.. Imnot saying that what you say is wrong, I just say: on Server Side all is working perfect. (All servers as well as Bungee running on the same OS/Machine/System)
And I tested it with some other players and everything works as pretended. No Problems at all.. and we switched servers during that test too.

Maybe couse of I set up the servers and Bungee, maybe becouse of the settings I made in Bungee and on the servers, maybe becouse of Fabric and the needed "FabricProxy" Mod on the Server... IDK
